src/js/service/ImportService.js

```diff
@@ -0,0 +1,121 @@
+/* Image and Animation import service supporting the import dialogs,
+ * file dropper service, and headless import API. */
+(function () {
+  /**
+   * @param {!PiskelController} piskelController
+   * @param {!PreviewController} previewController
+   * @constructor
+   */
+  var ImportService = $.namespace('pskl.service').ImportService =
+      function (piskelController, previewController) {
+    this.piskelController_ = piskelController;
+    this.previewController_ = previewController;
+  };
+
+  /**
+   * @param {!Image} image
+   * @param {!Object} options
+   * @param {!string} options.importType - 'single' if not spritesheet
+   * @param {!number} options.frameSizeX
+   * @param {!number} options.frameSizeY
+   * @param {number} [options.frameOffsetX] only used in spritesheet imports.
+   * @param {number} [options.frameOffsetY] only used in spritesheet imports.
+   * @param {!boolean} options.smoothing
+   * @param {function} [onComplete]
+   */
+  ImportService.prototype.newPiskelFromImage = function (image, options, onComplete) {
+    onComplete = onComplete || function () {};
```

Comment:
nit: you could use Constants.EMPTY_FUNCTION here (really not pushing for it, it's just FYI)